### PR TITLE
Stabilize AArch64 SHA3 intrinsics

### PR DESCRIPTION
**Pull request by tarcieri**
See also: rust-lang/rust#117225

### CODE REVIEW

File: crates/core_arch/src/aarch64/neon/generated.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ use stdarch_test::assert_instr;
 #[inline]
 #[target_feature(enable = "neon,sha3")]
 #[cfg_attr(test, assert_instr(eor3))]
-#[unstable(feature = "stdarch_neon_sha3", issue = "117225")]
+#[stable(feature = "stdarch_neon_sha3", since = "CURRENT_RUSTC_VERSION")]
```
Comment (bjorn3) [inline crates/core_arch/src/aarch64/neon/generated.rs:18]: Is `CURRENT_RUSTC_VERSION` supported in submodules?

Comment (tarcieri) [reply crates/core_arch/src/aarch64/neon/generated.rs:18]: I wasn't sure exactly what to put for this. `CURRENT_RUSTC_VERSION` was cargo culted from #1399, which performed a similar stabilization.

Comment (Amanieu) [reply crates/core_arch/src/aarch64/neon/generated.rs:18]: No, it's not supported. You should use the current nightly version.

**Comment by Amanieu** [comment]
These intrinsics are tested against clang and are known to produce the same results.

@rfcbot fcp merge

**Comment by Amanieu** [comment]
@rfcbot fcp merge

File: crates/core_arch/src/aarch64/neon/generated.rs (hunk 1 above)
Comment (tarcieri) [reply crates/core_arch/src/aarch64/neon/generated.rs:18]: Changed to `1.78.0` in 4bc5e32

**Comment by dtolnay** [comment]
@rfcbot reviewed

**Comment by BurntSushi** [comment]
The checkboxes are greyed out for me, but I approve.

**Comment by Amanieu** [comment]
I've ticked your box.

File: crates/core_arch/src/aarch64/neon/generated.rs [hunk 2]
```diff
@@ -15,7 +15,7 @@ use stdarch_test::assert_instr;
 #[inline]
 #[target_feature(enable = "neon,sha3")]
 #[cfg_attr(test, assert_instr(eor3))]
-#[unstable(feature = "stdarch_neon_sha3", issue = "117225")]
+#[stable(feature = "stdarch_neon_sha3", since = "1.78.0")]
```
Comment (tarcieri) [inline crates/core_arch/src/aarch64/neon/generated.rs:18]: @Amanieu do I need to bump this to 1.79.0 now?

Also seems like I need to merge/rebase?

Comment (Amanieu) [reply crates/core_arch/src/aarch64/neon/generated.rs:18]: You can do this when the FCP ends in 2 weeks.

**Comment by tarcieri** [comment]
Merging `master` caused some build failures which seem to be unrelated:

https://github.com/rust-lang/stdarch/actions/runs/8491609619/job/23263755908?pr=1552#step:14:8910

```
   Compiling intrinsic-test v0.1.0 (/checkout/crates/intrinsic-test)
error: field `reg` is never read
  --> crates/intrinsic-test/src/json_parser.rs:21:9
   |
19 |     Register {
   |     -------- field in this variant
20 |         #[serde(rename = "register")]
21 |         reg: String,
   |         ^^^
   |
   = note: `-D dead-code` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(dead_code)]`

error: could not compile `intrinsic-test` (bin "intrinsic-test") due to 1 previous error
Error: Process completed with exit code 101.
```

**Comment by Amanieu** [comment]
This seems to be fallout from https://github.com/rust-lang/rust/pull/119552 which was recently merged. Just add `#[allow(dead_code)]` on that field to ignore the lint.

File: crates/core_arch/src/aarch64/neon/generated.rs (hunk 2 above)
Comment (Amanieu) [reply crates/core_arch/src/aarch64/neon/generated.rs:18]: @tarcieri You should update the version to 1.79 now that FCP is done.

Comment (tarcieri) [reply crates/core_arch/src/aarch64/neon/generated.rs:18]: Done

**Comment by Amanieu** [comment]
Can you squash and rebase?

**Comment by tarcieri** [comment]
@Amanieu done

File: crates/intrinsic-test/src/json_parser.rs [hunk 3]
```diff
@@ -17,6 +17,7 @@ struct ReturnType {
 #[serde(untagged, deny_unknown_fields)]
 pub enum ArgPrep {
     Register {
+        #[allow(dead_code)]
         #[serde(rename = "register")]
         #[allow(dead_code)]
```
Comment (tarcieri) [inline crates/intrinsic-test/src/json_parser.rs:22]: @Amanieu just noticed a bit of a merge/rebase failure here, seems this attribute was duplicated

**Pull request merged**